Added a  test for mdamath.py

Pradyumn-cloud opened the pull request:
Fixes #5064 

Changes made in this Pull Request:

- Added a test file for mdamath.py
-

## PR Checklist
<!-- Please use the following checklist to ensure the PR is ready to be reviewed/merged. -->
 - [x] Issue raised/referenced?
 - [x] Tests updated/added?
 - [x] Documentation updated/added?




I certify that I can submit this code contribution as described in the [**Developer Certificate of Origin**](https://developercertificate.org/), under the MDAnalysis [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE).


<!-- readthedocs-preview mdanalysis start -->
----
📚 Documentation preview 📚: https://mdanalysis--5096.org.readthedocs.build/en/5096/

<!-- readthedocs-preview mdanalysis end -->

Pradyumn-cloud committed:
Added a  test for mdamath

orbeckst commented on testsuite/MDAnalysisTests/lib/test_mdamath.py:
The black-formatter check is failing (see https://github.com/MDAnalysis/mdanalysis/actions/runs/16780330516/job/47516944829?pr=5096#step:5:225 ): Please format with black 24 (see also https://userguide.mdanalysis.org/stable/contributing_code.html#code-formatting-in-python). Only format the files that you touched, i.e.,
```
black test_mdamath.py
```

orbeckst replied:
Still failing. Look at https://github.com/MDAnalysis/mdanalysis/actions/runs/16809252508/job/47609959096?pr=5096 and check which files black _would_ want to change.

orbeckst reviewed (changes requested):
Good start! Please also _remove_ the functions that you moved to your new files from test_util. Make sure that you're removing the `import lib.mdamath` statement at the top, too.

Make sure that your code is [correctly formatted](https://userguide.mdanalysis.org/stable/contributing_code.html#code-formatting-in-python) and passes all tests.

Pradyumn-cloud committed:
Updates

Formatted test_mdamath.py using black24
Updated the test_util.py as mentioned

Pradyumn-cloud committed:
Removing Errors

Updated the test_mdamath.py and test_util.py

orbeckst commented on testsuite/MDAnalysisTests/lib/test_util.py:
```diff
@@ -43,6 +43,7 @@
 
 import MDAnalysis as mda
 import MDAnalysis.lib.util as util
+import MDAnalysis.lib.mdamath as mdamath
```
remove

Pradyumn-cloud replied:
I think TestMakeWhole class uses mdamath in test_util.py

Pradyumn-cloud replied:
What would you like me to do?

orbeckst replied:
mdamath.make_util is a mdamath function so its tests should also be moved.

Looking at test_util.py, I don't see any other mdamath functions in there, so it should all work when the import line is removed.

Pradyumn-cloud committed:
Black error fix

Pradyumn-cloud committed:
Move make_whole tests to test_mdamath.py

Relocated the TestMakeWhole test class from test_util.py to test_mdamath.py for better organization and relevance. This change helps keep tests for mdamath functions within their dedicated test module.

Pradyumn-cloud committed:
linting error resolved

Pradyumn-cloud commented:
Can you provide further review.

RMeli commented:
/azp run

orbeckst reviewed (approved):
This looks all good. Thank you for the refactor.

Let's just wait for all the checks to pass.

orbeckst commented:
@RMeli thank you for also having a look. Please feel free to merge if/when you feel it's ready. I'm busy and might or might not be able to get to it today.

orbeckst commented:
Thank you for your contribution @Pradyumn-cloud 🎉 !!
